### PR TITLE
Rename multi-tab setting to synchronizeTabs

### PR DESCRIPTION
**Pull request by schmidt-sebastian**
Note: To be merged into next major release.

### CODE REVIEW

File: packages/firestore/src/api/database.ts [hunk 1]
```diff
@@ -398,14 +398,30 @@ export class Firestore implements firestore.FirebaseFirestore, FirebaseService {
...
+            '`synchronizeTabs`. In a future release, the setting  will be removed ' +
+            'and it is recommended that you remove it from your ' +
+            'firestore.enablePersistence() call now.'
+        );
+      }
+      synchronizeTabs = objUtils.defaulted(
+        settings.synchronizeTabs !== undefined
+          ? settings.synchronizeTabs
+          : settings.experimentalTabSynchronization,
+        DEFAULT_SYNCHRONIZE_TABS
```
Comment (mikelehen) [inline packages/firestore/src/api/database.ts]: Way to go naming the default correctly ahead of time!

File: packages/firestore/src/api/database.ts [hunk 2]
```diff
@@ -398,14 +398,30 @@ export class Firestore implements firestore.FirebaseFirestore, FirebaseService {
           'any other methods on a Firestore object.'
       );
     }
+
+    let synchronizeTabs = false;
+
+    if (settings) {
+      if (settings.experimentalTabSynchronization !== undefined) {
+        log.error(
+          'The `experimentalTabSynchronization` setting has been renamed to ' +
```
Comment (mikelehen) [inline packages/firestore/src/api/database.ts]: This is probably super nitty but I don't think we usually use backticks within log messages.  I'd remove or replace with `'`

File: packages/firestore/src/api/database.ts [hunk 3]
```diff
@@ -398,14 +398,30 @@ export class Firestore implements firestore.FirebaseFirestore, FirebaseService {
...
+            '`synchronizeTabs`. In a future release, the setting  will be removed ' +
```
Comment (mikelehen) [inline packages/firestore/src/api/database.ts]: extra space in `setting  will`

File: packages/firestore/src/api/database.ts [hunk 4]
```diff
@@ -398,14 +398,30 @@ export class Firestore implements firestore.FirebaseFirestore, FirebaseService {
...
+            'and it is recommended that you remove it from your ' +
```
Comment (mikelehen) [inline packages/firestore/src/api/database.ts]: I might be overly paranoid but I worry "the setting will be removed" and "recommend that you remove it" could potentially be ambiguous and lead people to remove the setting entirely.  Perhaps:

> In a future release, 'experimentalTabSynchronization' will be removed, so you should update your firestore.enablePersistence() call to use 'synchronizeTabs' now.

File: packages/firestore/src/api/database.ts (hunk 2 above)
Comment (schmidt-sebastian) [reply packages/firestore/src/api/database.ts]: I didn't like the message without any quoting, but I am fine just using single quotes.

File: packages/firestore/src/api/database.ts (hunk 3 above)
Comment (schmidt-sebastian) [reply packages/firestore/src/api/database.ts]: Ugh, I thought I fixed this before sending this out. It's fixed now.

File: packages/firestore/src/api/database.ts (hunk 4 above)
Comment (schmidt-sebastian) [reply packages/firestore/src/api/database.ts]: Sure, that works and reduces code duplication :)

File: packages/firestore/src/api/database.ts (hunk 1 above)
Comment (schmidt-sebastian) [reply packages/firestore/src/api/database.ts]: Why do you assume the API proposal wasn't based on this name? :)

**Comment by schmidt-sebastian** [comment]
Back to you Mike for merging! Thanls

**Comment by mikelehen** [comment]
cc/ @Feiyang1 Another change for the next release...

**Pull request merged**